medmodels/treatment_effect_estimation/treatment_effect.py

```diff
@@ -0,0 +1,512 @@
+            if criteria_filter:
+                self.treatment_all.update(
+                    set(
+                        self.medrecord.select_nodes(
+                            node()
+                            .index()
+                            .is_in(self.medrecord.group(self.patients_group))
+                            & criteria_filter
+                            & node().index().is_in(self.medrecord.neighbors(treatment))
+                        )
+                    )
+                )
+            else:
+                self.treatment_all.update(
+                    set(
+                        self.medrecord.select_nodes(
+                            node()
+                            .index()
+                            .is_in(self.medrecord.group(self.patients_group))
+                            & node().index().is_in(self.medrecord.neighbors(treatment))
+                        )
+                    )
+                )
+
+        assert (
+            self.treatment_all != set()
+        ), "No patients found for the treatment groups in this MedRecord"
+
+        # Finding the patients that had the outcome after the treatment
+        for outcome in self.outcomes:
+            nodes_to_check = self.medrecord.select_nodes(
+                node().index().is_in(self.medrecord.neighbors(outcome))
+                & node().index().is_in(list(self.treatment_all))
+            )
+            self.treatment_true.update(
+                {
+                    node_id
+                    for node_id in nodes_to_check
+                    if self._is_outcome_after_treatment(
+                        node_id, outcome, max_time_in_years
+                    )
+                }
+            )
+        self.treatment_false = self.treatment_all - self.treatment_true
+
+        # Find the controls (patients that did not undergo the treatment)
+        self.control_true, self.control_false = self.find_controls(
+            criteria_filter=criteria_filter,
+        )
+        control_all = self.control_true | self.control_false
+
+        # Matching control groups to treatment groups using the given matching class
+        if matching_class:
+            matching_instance = matching_class(
+                treated_group=self.treatment_all,
+                control_group=control_all,
+                medrecord=self.medrecord,
+                **matching_args,
+            )
+            matched_control = matching_instance.matched_control
+            self.control_true = matched_control & self.control_true
+            self.control_false = matched_control & self.control_false
+
+        # The groups have been sorted and the other methods can be called
+        self.groups_sorted = True
+
+    def _is_outcome_after_treatment(
+        self, node_id: NodeIndex, outcome: NodeIndex, max_time_in_years: float
+    ) -> bool:
+        """
+        Determines whether an outcome occurred after treatment within a specified time
```

Comment:
These private functions do not need docstrings as well IMO
(also just for reference, no change required)

Reply:
That is completely fair and super nice! Just wanted to say that you are not required to do so but if you want to, go ahead! :)